manual/src/refman/extensions/bindingops.etex

```diff
@@ -148,3 +139,57 @@ end
 
 where "(let*)" is bound to the "bind" operation, and "(and*)" is also
 bound to the monoidal product operation.
+
+\subsection{ss:letop-rules}{General desugaring rules}
+
+The form
+
+\begin{verbatim}
+let<op0>
+  x1 = e1
+and<op1>
+  x2 = e2
+and<op2>
+  x3 = e3
+in e
+\end{verbatim}
+
+desugars into
+
+\begin{verbatim}
+( let<op0> )
+  (( and<op2> )
+    (( and<op1> )
+      e1
+      e2)
+    e3)
+  (fun ((x1, x2), x3) -> e)
+\end{verbatim}
+
+This of course works for any number of nested and-operators, but the
+fully general rule is inconvenient to express and hard to read.
```

Comment:
Not suggesting a change here, but for the record when I explain the rewriting I normally utilize the following two "small-step" rules:

- `letOP x1 = e1 andOP x2 = e2 andOP ... in e ==> letOP (x1, x2) = (andOP) e1 e2 andOP ... in e`
- `letOP x1 = e1 in e ==> (letOP) e1 (fun x1 -> e)`

Reply:
Clever! You should write more PL papers :-) I had tried to provide a compositional rule turning bindings into functions (my idea was that `p = e` desugared `fun p -> e`, and `<bindings> and<op> <binding>` would be defined in terms of the desugaring of `<binding>` and `<bindings>`), and that did not work. I will use your suggestion.

Reply:
@nojb I included these desugaring rules in a revised version of the PR.